build-logic/convention/src/main/kotlin/AndroidJacocoConventionPlugin.kt

```diff
@@ -0,0 +1,56 @@
+import com.android.build.api.dsl.CommonExtension
+import com.android.build.api.dsl.LibraryExtension
+import com.android.build.api.variant.AndroidComponentsExtension
+import com.android.build.api.variant.ApplicationAndroidComponentsExtension
+import com.android.build.api.variant.LibraryAndroidComponentsExtension
+import com.android.build.gradle.internal.dsl.BaseAppModuleExtension
+import com.google.samples.apps.nowinandroid.configureJacoco
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.the
+
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+class AndroidJacocoConventionPlugin : Plugin<Project> {
+    override fun apply(target: Project) {
+        with(target) {
+            pluginManager.apply("jacoco")
+
+            val jacocoExtension: AndroidComponentsExtension<*, *, *>
+            val androidExtension: CommonExtension<*, *, *, *, *, *> = when {
```

Comment:
Please refactor this so that the `when` block sets both `androidExtension` and `jacocoExtension` and does not return anything. The current approach assigns a value to `androidExtension` but has a side effect of setting the `jacocoExtension`. Side effects should be avoided wherever possible.

Reply:
@dturner 
I refactored it as in https://github.com/android/nowinandroid/pull/1519/commits/2d066415f0c12a45c09359a049208cdc47241ebe

Is there anything or better ways I am supposed to do, kindly leave a comment me.
Thank you.

Reply:
I updated the code.

Reply:
Thank you. Look good to me!